src/bundles/ipfs-provider.js

```diff
@@ -104,33 +104,49 @@ const readAPIAddressSetting = () => {
   return setting == null ? null : asAPIAddress(setting)
 }
 
-const asAPIAddress = (value) => asMultiaddress(value) || asURL(value)
+const asAPIAddress = (value) => asHttpClientOptions(value) || asMultiaddress(value) || asURL(value)
 
 /**
- * Attempts to turn cast given value into `URL` instance. Return either `URL`
- * instance or `null`.
+ * Attempts to turn cast given value into URL.
+ * Return either string instance or `null`.
  * @param {any} value
  * @returns {string|null}
  */
 const asURL = (value) => {
+  if (!value || typeof value !== 'string') return null
```

Comment:
NIT: I don't think it's really necessary, because URL will stringify and if it's not valid URL it will error anyway. Before this change passing a URL instance would have worked with this change it no longer will, I don't remember if that can be a problem.